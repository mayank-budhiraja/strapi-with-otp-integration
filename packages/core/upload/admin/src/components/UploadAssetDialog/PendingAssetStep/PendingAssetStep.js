import React, { useState, useRef } from 'react';
import PropTypes from 'prop-types';
import { ModalHeader, ModalBody, ModalFooter } from '@strapi/design-system/ModalLayout';
import { Typography } from '@strapi/design-system/Typography';
import { Button } from '@strapi/design-system/Button';
import { useIntl } from 'react-intl';
import { Flex } from '@strapi/design-system/Flex';
import { Stack } from '@strapi/design-system/Stack';
import { Grid, GridItem } from '@strapi/design-system/Grid';
import { KeyboardNavigable } from '@strapi/design-system/KeyboardNavigable';
import { AssetCard } from '../../AssetCard/AssetCard';
import { UploadingAssetCard } from '../../AssetCard/UploadingAssetCard';
import getTrad from '../../../utils/getTrad';
import { AssetDefinition } from '../../../constants';

const Status = {
  Idle: 'IDLE',
  Uploading: 'UPLOADING',
  Intermediate: 'INTERMEDIATE',
};

export const PendingAssetStep = ({
  addUploadedFiles,
  onClose,
  onEditAsset,
  onRemoveAsset,
  assets,
  onClickAddAsset,
  onCancelUpload,
  onUploadSucceed,
}) => {
  const assetCountRef = useRef(0);
  const { formatMessage } = useIntl();
  const [uploadStatus, setUploadStatus] = useState(Status.Idle);

  const handleSubmit = async e => {
    e.preventDefault();
    e.stopPropagation();

    setUploadStatus(Status.Uploading);
  };

  const handleStatusChange = (status, file) => {
    if (status === 'success' || status === 'error') {
      assetCountRef.current++;

      // There's no "terminated" status. When all the files have called their
      // onUploadSucceed callback, the parent component filters the asset list
      // and closes the modal when the asset list is empty
      if (assetCountRef.current === assets.length) {
        assetCountRef.current = 0;
        setUploadStatus(Status.Intermediate);
      }
    }

    if (status === 'success') {
      onUploadSucceed(file);
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <ModalHeader>
        <Typography fontWeight="bold" textColor="neutral800" as="h2" id="title">
          {formatMessage({
            id: getTrad('header.actions.add-assets'),
            defaultMessage: 'Add new assets',
          })}
        </Typography>
      </ModalHeader>

      <ModalBody>
        <Stack spacing={7}>
          <Flex justifyContent="space-between">
            <Stack spacing={0}>
              <Typography variant="pi" fontWeight="bold" textColor="neutral800">
                {formatMessage(
                  {
                    id: getTrad('list.assets.selected'),
                    defaultMessage:
                      '{number, plural, =0 {No asset} one {1 asset} other {# assets}} ready to upload',
                  },
                  { number: assets.length }
                )}
              </Typography>
              <Typography variant="pi" textColor="neutral600">
                {formatMessage({
                  id: getTrad('modal.upload-list.sub-header-subtitle'),
                  defaultMessage: 'Manage the assets before adding them to the Media Library',
                })}
              </Typography>
            </Stack>
            <Button size="S" onClick={onClickAddAsset}>
              {formatMessage({
                id: getTrad('header.actions.add-assets'),
                defaultMessage: 'Add new assets',
              })}
            </Button>
          </Flex>
          <KeyboardNavigable tagName="article">
            <Grid gap={4}>
              {assets.map(asset => {
                const assetKey = asset.url;

                if (uploadStatus === Status.Uploading || uploadStatus === Status.Intermediate) {
                  return (
                    <GridItem col={4} key={assetKey}>
                      <UploadingAssetCard
                        // Props used to store the newly uploaded files
                        addUploadedFiles={addUploadedFiles}
                        asset={asset}
                        id={assetKey}
                        onCancel={onCancelUpload}
                        onStatusChange={status => handleStatusChange(status, asset.rawFile)}
                        size="S"
                      />
                    </GridItem>
                  );
                }

                return (
                  <GridItem col={4} key={assetKey}>
                    <AssetCard
                      asset={asset}
                      size="S"
                      key={assetKey}
                      local
                      alt={asset.name}
                      onEdit={onEditAsset}
                      onRemove={onRemoveAsset}
                    />
                  </GridItem>
                );
              })}
            </Grid>
          </KeyboardNavigable>
        </Stack>
      </ModalBody>

      <ModalFooter
        startActions={
          <Button onClick={onClose} variant="tertiary">
            {formatMessage({ id: 'app.components.Button.cancel', defaultMessage: 'cancel' })}
          </Button>
        }
        endActions={
          <Button type="submit" loading={uploadStatus === Status.Uploading}>
            {formatMessage(
              {
                id: getTrad('modal.upload-list.footer.button'),
                defaultMessage:
                  'Upload {number, plural, one {# asset} other {# assets}} to the library',
              },
              { number: assets.length }
            )}
          </Button>
        }
      />
    </form>
  );
};

PendingAssetStep.defaultProps = {
  addUploadedFiles: undefined,
};

PendingAssetStep.propTypes = {
  addUploadedFiles: PropTypes.func,
  assets: PropTypes.arrayOf(AssetDefinition).isRequired,
  onClose: PropTypes.func.isRequired,
  onEditAsset: PropTypes.func.isRequired,
  onRemoveAsset: PropTypes.func.isRequired,
  onClickAddAsset: PropTypes.func.isRequired,
  onUploadSucceed: PropTypes.func.isRequired,
  onCancelUpload: PropTypes.func.isRequired,
};
