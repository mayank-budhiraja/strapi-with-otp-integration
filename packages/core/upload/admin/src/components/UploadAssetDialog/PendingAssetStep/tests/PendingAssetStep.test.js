import React from 'react';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { render as renderTL } from '@testing-library/react';
import { QueryClientProvider, QueryClient } from 'react-query';
import { PendingAssetStep } from '../PendingAssetStep';
import en from '../../../../translations/en.json';

jest.mock('../../../../utils/getTrad', () => x => x);
jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: jest.fn(({ id }) => en[id]) }),
}));

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
    },
  },
});

describe('PendingAssetStep', () => {
  it('snapshots the component with valid cards', () => {
    const assets = [
      {
        source: 'url',
        type: 'image',
        url: 'http://localhost:5000/CPAM.jpg',
        ext: 'jpg',
        mime: 'image/jpeg',
      },
      {
        source: 'url',
        type: 'doc',
        url: 'http://localhost:5000/MARIAGE%20FRACHET%204.pdf',
        ext: 'pdf',
        mime: 'application/pdf',
      },
      {
        source: 'url',
        type: 'video',
        url: 'http://localhost:5000/mov_bbb.mp4',
        ext: 'mp4',
        mime: 'video/mp4',
      },
    ];

    const { container } = renderTL(
      <QueryClientProvider client={queryClient}>
        <ThemeProvider theme={lightTheme}>
          <PendingAssetStep
            assets={assets}
            onClose={jest.fn()}
            onAddAsset={jest.fn()}
            onClickAddAsset={jest.fn()}
            onCancelUpload={jest.fn()}
            onUploadSucceed={jest.fn()}
          />
        </ThemeProvider>
      </QueryClientProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
