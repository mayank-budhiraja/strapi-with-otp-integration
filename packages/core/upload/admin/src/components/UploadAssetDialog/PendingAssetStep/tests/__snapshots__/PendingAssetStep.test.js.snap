// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingAssetStep snapshots the component with valid cards 1`] = `
.c53 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  background: #f6f6f9;
  padding-top: 16px;
  padding-right: 20px;
  padding-bottom: 16px;
  padding-left: 20px;
}

.c6 {
  padding-top: 24px;
  padding-right: 40px;
  padding-bottom: 24px;
  padding-left: 40px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c50 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c4 svg {
  height: 12px;
  width: 12px;
}

.c4 svg > g,
.c4 svg path {
  fill: #ffffff;
}

.c4[aria-disabled='true'] {
  pointer-events: none;
}

.c4:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c4:focus-visible {
  outline: none;
}

.c4:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
}

.c5 svg > g,
.c5 svg path {
  fill: #8e8ea9;
}

.c5:hover svg > g,
.c5:hover svg path {
  fill: #666687;
}

.c5:active svg > g,
.c5:active svg path {
  fill: #a5a5ba;
}

.c5[aria-disabled='true'] {
  background-color: #eaeaef;
}

.c5[aria-disabled='true'] svg path {
  fill: #666687;
}

.c1 {
  border-radius: 4px 4px 0 0;
  border-bottom: 1px solid #eaeaef;
}

.c49 {
  border-radius: 0 0 4px 4px;
  border-top: 1px solid #eaeaef;
}

.c51 > * + * {
  margin-left: 8px;
}

.c7 {
  overflow: auto;
  max-height: 60vh;
}

.c3 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c12 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c13 {
  color: #666687;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c17 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c14 svg {
  height: 12px;
  width: 12px;
}

.c14 svg > g,
.c14 svg path {
  fill: #ffffff;
}

.c14[aria-disabled='true'] {
  pointer-events: none;
}

.c14:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c14:focus-visible {
  outline: none;
}

.c14:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c15 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  background: #4945ff;
  border: 1px solid #4945ff;
}

.c15 .sc-hcupDf {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c15 .c16 {
  color: #ffffff;
}

.c15[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c15[aria-disabled='true'] .c16 {
  color: #666687;
}

.c15[aria-disabled='true'] svg > g,
.c15[aria-disabled='true'] svg path {
  fill: #666687;
}

.c15[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c15[aria-disabled='true']:active .c16 {
  color: #666687;
}

.c15[aria-disabled='true']:active svg > g,
.c15[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c15:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c15:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c15 svg > g,
.c15 svg path {
  fill: #ffffff;
}

.c52 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  background: #4945ff;
  border: 1px solid #4945ff;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c52 .sc-hcupDf {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c52 .c16 {
  color: #ffffff;
}

.c52[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c52[aria-disabled='true'] .c16 {
  color: #666687;
}

.c52[aria-disabled='true'] svg > g,
.c52[aria-disabled='true'] svg path {
  fill: #666687;
}

.c52[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c52[aria-disabled='true']:active .c16 {
  color: #666687;
}

.c52[aria-disabled='true']:active svg > g,
.c52[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c52:hover {
  background-color: #f6f6f9;
}

.c52:active {
  background-color: #eaeaef;
}

.c52 .c16 {
  color: #32324d;
}

.c52 svg > g,
.c52 svg path {
  fill: #32324d;
}

.c37 {
  width: 100%;
  height: 5.5rem;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c38 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c9 > * + * {
  margin-top: 32px;
}

.c11 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c18 {
  display: grid;
  grid-template-columns: repeat(12,1fr);
  gap: 16px;
}

.c19 {
  grid-column: span 4;
  max-width: 100%;
}

.c28 {
  padding-top: 4px;
}

.c44 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c20 {
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c25 {
  padding-top: 8px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 12px;
}

.c33 {
  background: #f6f6f9;
  padding: 4px;
  border-radius: 4px;
  min-width: 20px;
}

.c45 {
  background: #32324d;
  color: #ffffff;
  padding: 4px;
  border-radius: 4px;
}

.c21 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c26 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c34 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c42 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c24 {
  margin: 0;
  padding: 0;
  max-height: 100%;
  max-width: 100%;
  object-fit: contain;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5.5rem;
  width: 100%;
  background: repeating-conic-gradient(#f6f6f9 0% 25%,transparent 0% 50%) 50% / 20px 20px;
}

.c29 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c30 {
  color: #666687;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c36 {
  color: #666687;
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
}

.c47 {
  color: #ffffff;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c32 {
  margin-left: auto;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c35 {
  margin-left: 4px;
}

.c27 {
  word-break: break-all;
}

.c22 {
  position: relative;
  border-bottom: 1px solid #eaeaef;
}

.c46 {
  position: absolute;
  bottom: 4px;
  right: 4px;
}

.c31 {
  text-transform: uppercase;
}

.c48 {
  text-transform: uppercase;
}

.c43 canvas,
.c43 video {
  display: block;
  max-width: 100%;
  max-height: 5.5rem;
}

.c41 {
  text-transform: uppercase;
}

.c40 svg {
  font-size: 3rem;
}

.c39 {
  border-radius: 4px 4px 0 0;
  background: linear-gradient(180deg,#ffffff 0%,#f6f6f9 121.48%);
}

@media (max-width:68.75rem) {
  .c19 {
    grid-column: span;
  }
}

@media (max-width:34.375rem) {
  .c19 {
    grid-column: span;
  }
}

<div>
  <form>
    <div
      class="c0 c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
          id="title"
        >
          Add new assets
        </h2>
        <button
          aria-disabled="false"
          aria-label="Close the modal"
          class="c4 c5"
          type="button"
        >
          <svg
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 2.417L21.583 0 12 9.583 2.417 0 0 2.417 9.583 12 0 21.583 2.417 24 12 14.417 21.583 24 24 21.583 14.417 12 24 2.417z"
              fill="#212134"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c6 c7"
    >
      <div
        class="c8 c9"
        spacing="7"
      >
        <div
          class="c10"
        >
          <div
            class="c8 c11"
          >
            <span
              class="c12"
            >
              {number, plural, =0 {No asset} one {1 asset} other {# assets}} ready to upload
            </span>
            <span
              class="c13"
            >
              Manage the assets before adding them to the Media Library
            </span>
          </div>
          <button
            aria-disabled="false"
            class="c14 c15"
            type="button"
          >
            <span
              class="c16 c17"
            >
              Add new assets
            </span>
          </button>
        </div>
        <div
          class=""
        >
          <div
            class="c18"
          >
            <div
              class="c19"
            >
              <div
                class=""
              >
                <article
                  aria-labelledby="card-1-title"
                  class="c20"
                  tabindex="0"
                >
                  <div
                    class="c21 c22"
                  >
                    <div
                      class="c23"
                    >
                      <img
                        aria-hidden="true"
                        class="c24"
                        src="http://localhost:5000/CPAM.jpg"
                      />
                    </div>
                  </div>
                  <div
                    class="c25"
                  >
                    <div
                      class="c26"
                    >
                      <div
                        class="c27"
                      >
                        <div
                          class="c28"
                        >
                          <h2
                            class="c29"
                            id="card-1-title"
                          />
                        </div>
                        <div
                          class="c30"
                        >
                          <span
                            class="c31"
                          >
                            jpg
                          </span>
                        </div>
                      </div>
                      <div
                        class="c32"
                      >
                        <div
                          class="c33 c34 c35"
                        >
                          <span
                            class="c36"
                          >
                            Image
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </article>
              </div>
            </div>
            <div
              class="c19"
            >
              <div
                class=""
              >
                <article
                  aria-labelledby="card-2-title"
                  class="c20"
                  tabindex="0"
                >
                  <div
                    class="c21 c22"
                  >
                    <div
                      class="c37 c38 c39"
                      height="5.5rem"
                      width="100%"
                    >
                      <span
                        class="c40"
                      >
                        <svg
                          fill="none"
                          height="1em"
                          viewBox="0 0 24 33"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M23.305 8.126L16.39.749c-.443-.472-1.043-.741-1.643-.741H2.337c-.632 0-1.2.27-1.642.74A2.529 2.529 0 000 2.5v27.015c0 1.381 1.042 2.493 2.337 2.493h19.326c1.295 0 2.337-1.112 2.337-2.493V9.877c0-.64-.253-1.28-.695-1.751zm-1.326.606h-5.116c-.568 0-1.042-.505-1.042-1.112V2.197l6.158 6.535zm-.316 21.827H2.337c-.537 0-.98-.471-.98-1.044V2.5c0-.269.096-.538.285-.74a.949.949 0 01.695-.304h12.126V7.62c0 1.415 1.074 2.56 2.4 2.56h5.78v19.335c0 .573-.443 1.044-.98 1.044z"
                            fill="#D9822F"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M13.964 20.095c.344.252.781.522 1.312.811a16.714 16.714 0 011.827-.1c1.53 0 2.451.236 2.764.709.166.212.177.463.031.753 0 .01-.005.02-.016.029l-.03.029v.014c-.063.367-.433.55-1.11.55-.5 0-1.098-.096-1.795-.29a11.908 11.908 0 01-2.03-.767c-2.3.232-4.341.633-6.121 1.202-1.593 2.53-2.852 3.795-3.779 3.795a.965.965 0 01-.437-.102l-.375-.174a1.668 1.668 0 00-.093-.072c-.105-.096-.136-.27-.094-.521.094-.386.385-.828.874-1.325.49-.498 1.177-.963 2.061-1.398.146-.087.266-.058.36.087.02.02.03.039.03.058a34.703 34.703 0 001.672-2.853c.707-1.313 1.249-2.578 1.623-3.794a10.952 10.952 0 01-.476-2.31c-.067-.748-.034-1.364.102-1.847.114-.386.333-.579.655-.579h.344c.24 0 .422.072.547.217.187.203.234.531.14.985a.307.307 0 01-.062.116.35.35 0 01.015.116v.434c-.02 1.188-.093 2.115-.218 2.78.572 1.584 1.332 2.733 2.28 3.447zM4.97 26.047c.542-.232 1.255-.995 2.14-2.289-.531.387-.987.792-1.367 1.217-.38.425-.637.782-.773 1.072zm6.183-11.412c-.135-.868-.125-1.506.032-1.911v-.03c.01-.009.015-.019.015-.028a.798.798 0 01.203.521c0 .01.005.024.016.044 0 .01.005.019.015.029a.317.317 0 00-.062.116c-.073.386-.11.593-.11.622l-.109.637zm-1.904 7.66a24.288 24.288 0 014.434-1.173 2.398 2.398 0 01-.203-.137 2.761 2.761 0 01-.25-.196c-.79-.647-1.452-1.496-1.983-2.549-.28.83-.713 1.782-1.296 2.854-.312.54-.546.94-.702 1.201zm7.9-.58c1.208 0 1.937.117 2.187.349.02.019.031.033.031.043-.042.01-.135.014-.281.014-.5 0-1.145-.135-1.936-.405z"
                            fill="#D9822F"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                  <div
                    class="c25"
                  >
                    <div
                      class="c26"
                    >
                      <div
                        class="c27"
                      >
                        <div
                          class="c28"
                        >
                          <h2
                            class="c29"
                            id="card-2-title"
                          />
                        </div>
                        <div
                          class="c30"
                        >
                          <span
                            class="c41"
                          >
                            pdf
                          </span>
                        </div>
                      </div>
                      <div
                        class="c32"
                      >
                        <div
                          class="c33 c34 c35"
                        >
                          <span
                            class="c36"
                          >
                            Doc
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </article>
              </div>
            </div>
            <div
              class="c19"
            >
              <div
                class=""
              >
                <article
                  aria-labelledby="card-3-title"
                  class="c20"
                  tabindex="0"
                >
                  <div
                    class="c21 c22"
                  >
                    <div
                      class="c23"
                    >
                      <div
                        class="c42"
                      >
                        <div
                          class="c43"
                        >
                          <figure
                            class=""
                          >
                            <video
                              crossorigin="anonymous"
                              src="http://localhost:5000/mov_bbb.mp4"
                            >
                              <source
                                type="video/mp4"
                              />
                            </video>
                            <figcaption
                              class="c44"
                            />
                          </figure>
                        </div>
                      </div>
                    </div>
                    <time
                      class="c45 c46"
                    >
                      <span
                        class="c47"
                      >
                        ...
                      </span>
                    </time>
                  </div>
                  <div
                    class="c25"
                  >
                    <div
                      class="c26"
                    >
                      <div
                        class="c27"
                      >
                        <div
                          class="c28"
                        >
                          <h2
                            class="c29"
                            id="card-3-title"
                          />
                        </div>
                        <div
                          class="c30"
                        >
                          <span
                            class="c48"
                          >
                            mp4
                          </span>
                        </div>
                      </div>
                      <div
                        class="c32"
                      >
                        <div
                          class="c33 c34 c35"
                        >
                          <span
                            class="c36"
                          >
                            Video
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c0 c49"
    >
      <div
        class="c2"
      >
        <div
          class="c50 c51"
        >
          <button
            aria-disabled="false"
            class="c14 c52"
            type="button"
          >
            <span
              class="c16 c17"
            />
          </button>
        </div>
        <div
          class="c50 c51"
        >
          <button
            aria-disabled="false"
            class="c14 c15"
            type="submit"
          >
            <span
              class="c16 c17"
            >
              Upload {number, plural, one {# asset} other {# assets}} to the library
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
  <div
    class="c53"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
